nixos/modules/services/hardware/keyd.nix

```diff
@@ -0,0 +1,68 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta = {
+    maintainers = [ "cidkid" ];
+  };
+
+  options = {
+    services.keyd = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          enable keyd daemon for remapping keys
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.keyd;
+        defaultText = literalExpression "pkgs.keyd";
+        description = ''
+          KeyD Package to use
+        '';
+      };
+
+      configuration = mkOption {
+        type = types.attrsOf (types.submodule ({ config, options, ... }: {
+          options = {
+            text = mkOption {
+              type = types.lines;
+              default = null;
+            };
+          };
+        }));
+
+        default = {};
+
+        description = ''
+          What the filename should be for configuration files for keyd (Multiple are allowed in keyd)
+        '';
+      };
+    };
+  };
+
+  config = mkIf config.services.keyd.enable {
```

Comment:
I'd suggest to extract the `config.services.<foo>` prefix to a let binding (i.e. `let cfg = config.services.keyd; in`) to avoid unnecessary repetition (Many other modules do the same).

Reply:
fixed